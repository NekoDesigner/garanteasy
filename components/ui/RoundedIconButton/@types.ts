/**
 * RoundedIconButton - Generated by Garanteasier CLI
 */

export interface IRoundedIconButtonProps {
  style?: object;
  testID?: string;
  icon: IRoundedIconButtonType;
  onPress: () => void;
}

export type IRoundedIconButtonType = 'scan' | 'settings' | 'arrow-left' | 'arrow-right'
