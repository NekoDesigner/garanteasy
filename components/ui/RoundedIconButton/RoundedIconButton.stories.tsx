/**
 * RoundedIconButton - Generated by Garanteasier CLI
 */

import React from "react";
import { View } from "react-native";
import RoundedIconButton from "./index";

export default {
  title: 'RoundedIconButton',
  component: RoundedIconButton
};

export const Default = () => (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center', padding: 16, gap: 24 }}>
      <RoundedIconButton icon="arrow-left" onPress={() => {}} />
    </View>
);