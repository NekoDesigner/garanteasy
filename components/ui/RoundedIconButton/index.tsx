/**
 * RoundedIconButton - Generated by Garanteasier CLI
 */

import React from "react";
import { Text, TouchableOpacity } from "react-native";
import { IRoundedIconButtonProps } from "./@types";
import RoundedIconButtonStyles from "./styles";

const RoundedIconButton: React.FC<IRoundedIconButtonProps> = ({ style, testID = 'roundediconbutton', ...props }) => {
  function getIconComponent(icon: IRoundedIconButtonProps['icon']) {
    switch (icon) {
      case 'scan':
        return <Text>Scan Icon</Text>;
      case 'settings':
        return <Text>Settings Icon</Text>;
      case 'arrow-left':
        return <Text>Arrow Left Icon</Text>;
      case 'arrow-right':
        return <Text>Arrow Right Icon</Text>;
      default:
        return <Text>Default Icon</Text>;
    }
  }
  return (
    <TouchableOpacity style={[RoundedIconButtonStyles, style]} testID={testID} {...props}>
      {getIconComponent(props.icon)}
    </TouchableOpacity>
  );
};

export default RoundedIconButton;
