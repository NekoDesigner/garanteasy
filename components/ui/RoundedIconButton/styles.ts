/**
 * {{ElementComponent}} - Generated by Garanteasier CLI
 */

import { StyleSheet } from 'react-native';
import { COLORS, SIZES } from '../../../constants';
import { IRoundedIconButtonSize } from './@types';

function getSize(size: IRoundedIconButtonSize): number {
  if (size === 'normal') {
    return 40;
  }
if (size === 'small') {
    return 32;
  }
  if (size === 'big') {
    return 70;
  }
  return 40; // Default case
}

const styles = function getRoundedIconButtonStyle(size: IRoundedIconButtonSize) {
  return StyleSheet.create({
    container: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      flexDirection: 'column',
      backgroundColor: COLORS.primary,
      borderRadius: SIZES.radius.full,
      height: getSize(size),
      width: getSize(size),
    }
  });
};

export default styles;
