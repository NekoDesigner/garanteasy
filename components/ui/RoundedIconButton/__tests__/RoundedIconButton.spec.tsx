/**
 * RoundedIconButton - Generated by Garanteasier CLI
 */

import { render } from "@testing-library/react-native";
import React from "react";
import RoundedIconButton from "../index";

describe("RoundedIconButton Components", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

    test('renders correctly', () => {
      const tree = render(<RoundedIconButton />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});