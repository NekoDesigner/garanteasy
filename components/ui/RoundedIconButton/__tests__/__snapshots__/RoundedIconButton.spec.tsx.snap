// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedIconButton Components renders correctly 1`] = `
<View
  style={
    [
      {},
      undefined,
    ]
  }
  testID="roundediconbutton"
>
  <Text>
    RoundedIconButton
  </Text>
</View>
`;
